### ISSUE TITLE
NPE when starting a repl

### ISSUE THREAD

**Issue by lassemaatta** [thheller/shadow-cljs#360]
After nrepl 0.4 support landed in `2.4.25`, we've had problems starting a repl. We use `lein` and have our `:nrepl-middleware`'s configured 1:1 with the user guide. 

Example output: https://gist.github.com/lassemaatta/33c8e6e42e81cdc43727402e06180b0d

Are we doing something silly or what might be the issue? I can try to provide an example project if needed.

**Comment by lassemaatta** [comment]
I think I narrowed it down after a bunch of trial and error: try including a namespace which refers to `shadow.cljs.devtools.server :as server`. For example I had:

```
(ns user
  (:require [shadow.cljs.devtools.server :as server]))

(defn go []
  (server/start!))
```

**Comment by thheller** [comment]
Ah right. `user.clj` is loaded before anything else so the check I have looking for `nrepl` 0.4 happens at an unexpected time.

You can quickfix it by requiring `clojure.tools.nrepl.server` before anything else.

```
(ns user
  (:require
   [clojure.tools.nrepl.server]
   [shadow.cljs.devtools.server :as server]))
```

I'll see if I can make the detection a bit more robust.

**Comment by lassemaatta** [comment]
Great, thanks!

**Comment by thheller** [comment]
This should be fixed as of `2.4.32`.
